### ISSUE TITLE
pod install error: minumum deployment target mismatch

### ISSUE THREAD

**Issue by ryan-cha** [CocoaPods/pod-template#264]
Hello! 

It seems that the latest [merge](https://github.com/CocoaPods/pod-template/commit/313e000e4852c526aad01e44c429d2fa5421a789) is breaking something.

Creating lib with demo app works okay for macOS platform, but **NOT okay for iOS platform**. 
Specifically, `pod install` has problem.

* pod version: `1.11.3`

* How to produce the error
  1. In terminal, run `pod lib create {libname}`
  2. Answer in order of `iOS`, `Swift`, `Yes`, `None`, `No`

* Expected behavior
  - The local library pod is successfully installed in the demo application
  - XCode opens automatically

* Actual behavior
  - The local library pod cannot be installed in the demo application.

Following is the copy of terminal output. Will submit PR soon. 

```

$ pod --version
1.11.3

$ pod lib create mylib

Cloning `https://github.com/CocoaPods/pod-template.git` into `mylib`.
Configuring mylib template.

------------------------------

To get you started we need to ask a few questions, this should only take a minute.

If this is your first time we recommend running through with the guide:
 - https://guides.cocoapods.org/making/using-pod-lib-create.html
 ( hold cmd and click links to open in a browser. )


What platform do you want to use?? [ iOS / macOS ]
 > iOS

What language do you want to use?? [ Swift / ObjC ]
 > Swift

Would you like to include a demo application with your library? [ Yes / No ]
 > Yes

Which testing frameworks will you use? [ Quick / None ]
 > None

Would you like to do view based testing? [ Yes / No ]
 > No
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint:
hint: 	git branch -m <name>

Running pod install on your new library.

Analyzing dependencies
[!] CocoaPods could not find compatible versions for pod "mylib":
  In Podfile:
    mylib (from `../`)

Specs satisfying the `mylib (from `../`)` dependency were found, but they required a higher minimum deployment target.

 Ace! you're ready to go!
 We will start you off by opening your project in Xcode
  open 'mylib/Example/mylib.xcworkspace'
The file /Users/ryancha/dev/personal/Android/mylibMobileFinal/lib/mylib/Example/mylib.xcworkspace does not exist.

To learn more about the template see `https://github.com/CocoaPods/pod-template.git`.
To learn more about creating a new pod, see `https://guides.cocoapods.org/making/making-a-cocoapod`.
```

**Comment by orta** [comment]
Might need a corrseponding change in the root xcodeproject to bump to ios 10?

**Comment by ryan-cha** [comment]
Fixed by #265
